### PR TITLE
javax -> jakarta. packages and namespace 

### PR DESCRIPTION
**Pull request by thadumi**
Hi @scottkurz ,
I used your suggestion creating separated commits: one for the with-spaces cleanup, one for the packages migration, and one for the namespace update.

### CODE REVIEW

File: com.ibm.jbatch.tck.dist.exec/tck.execution.for.ri.properties [hunk 1]
```diff
@@ -52,17 +46,13 @@ jvm.options=-Dcom.ibm.jbatch.spi.ServiceRegistry.BATCH_THREADPOOL_SERVICE=com.ib
...
 sigtestdev.classes=../runSigTest/sigtestdev.jar
-
 # The Batch API being tested
 batch.api.classes=../runAgainstRI/jakarta.batch-api.jar
-
 # shipped by TCK
-javax.inject.classes=lib/jakarta.inject-api-1.0.jar
-javax.enterprise.util.classes=lib/jakarta.enterprise.cdi-api-2.0.1.jar
+jakarta.inject.classes=lib/jakarta.inject-api-1.0.jar
+jakarta.enterprise.util.classes=lib/jakarta.enterprise.cdi-api-2.0.1.jar
```
Comment (scottkurz) [inline com.ibm.jbatch.tck.dist.exec/tck.execution.for.ri.properties:58]: These are fine for this PR.   Just observing they'll have to change before we can actually run anything, since we need these dependencies at a level with the jakarta.* packages.   I know these originally come from one of the POMs too.

File: com.ibm.jbatch.tck.ann.proc/pom.xml [hunk 2]
```diff
@@ -17,33 +17,34 @@
...
-  </build>
-  
-  <dependencies>
-    <dependency>
-      <groupId>jakarta.batch</groupId>
-      <artifactId>com.ibm.jbatch.tck.ann</artifactId>
-    </dependency>
-  </dependencies>
-  
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
```
Comment (scottkurz) [inline com.ibm.jbatch.tck.ann.proc/pom.xml:20]: Maybe before we're all done we can go back and update the copyright in all the files (or it could be another PR).  Something like:

> Copyright 2012, 2020 International Business Machines Corp. and others

File: com.ibm.jbatch.tck.exec/default.tck.exec.properties [hunk 3]
```diff
@@ -29,27 +28,21 @@ com.ibm.jbatch.spi.ServiceRegistry.BATCH_THREADPOOL_SERVICE=com.ibm.jbatch.conta
 com.ibm.jbatch.spi.ServiceRegistry.CONTAINER_ARTIFACT_FACTORY_SERVICE=com.ibm.jbatch.container.services.impl.WeldSEBatchArtifactFactoryImpl
 #com.ibm.jbatch.spi.ServiceRegistry.DELEGATING_ARTIFACT_FACTORY_SERVICE=com.ibm.jbatch.container.services.impl.DelegatingBatchArtifactFactoryImpl
 com.ibm.jbatch.spi.ServiceRegistry.J2SE_MODE=true
-
 # Test Class: ExecutionTests
 ExecutionTests.testInvokeJobWithUserStop.sleep=1000
```
Comment (scottkurz) [inline com.ibm.jbatch.tck.exec/default.tck.exec.properties:32]: You don't like the spacing in the properties files?    I thought it was a bit more readable but not too attached either way.

File: jakarta.batch.official.tck/src/main/asciidoc/batch-tck-reference-guide.adoc [hunk 4]
```diff
@@ -289,34 +315,42 @@ The list of test cases to run can be customized (for debugging) by modifying the
...
+
+== Working with TCK source (debugging, etc.)
 
 For most development/debug use cases it is recommended to refer to the source in the Jakarta Batch TCK] GitHub repository [https://github.com/eclipse-ee4j/batch-tck], and to leverage the Maven automation and artifacts there using the associated documentation.
 
 It should be documented how to use tags/releases, etc. to match the official level tested in the TCK distribution.
 
-It is also possible to use the TestNG `build.xml` script's *compile* target, after setting the *src* property appropriately.   We have paid less attention to this more recently and instead focused on the Maven approach.
+It is also possible to use the TestNG `build.xml` script's *compile* target, after setting the *src* property appropriately.
+We have paid less attention to this more recently and instead focused on the Maven approach.
```
Comment (scottkurz) [inline jakarta.batch.official.tck/src/main/asciidoc/batch-tck-reference-guide.adoc:349]: Is making a new paragraph just a personal preference  or is this part of some well-known style or formatting you applied?

File: jakarta.batch.official.tck/src/main/asciidoc/batch-tck-reference-guide.adoc [hunk 5]
```diff
@@ -289,34 +315,42 @@ The list of test cases to run can be customized (for debugging) by modifying the
...
+The test also can't end too soon, causing a test failure because the wait was not long enough.
 
 This timeout value can be customized (say, to increase when debugging or decrease to force a faster failure in some cases).
 
 Note that some of the tests (e.g. the chunk tests involving time-based checkpointing) will take at least 15-25 seconds to run on any hardware, so any value less than that for the whole TCK will cause some test failures simply due to timing (and not because of any failure in the underlying Jakarta Batch implementation).
 
-The 900 seconds value, then, was chosen to avoid falsely reporting an error because of timing out too soon, allowing plenty of leeway.	It also facilitates debugging.	It does not, however, provide â€œfast failureâ€ in case of a hang or runaway thread.
+The 900 seconds value, then, was chosen to avoid falsely reporting an error because of timing out too soon, allowing plenty of leeway.
+It also facilitates debugging.
+It does not, however, provide â€œfast failureâ€ in case of a hang or runaway thread.
```
Comment (scottkurz) [inline jakarta.batch.official.tck/src/main/asciidoc/batch-tck-reference-guide.adoc:334]: Probably what is `â€œfast failureâ€ ` was once intended to be `"fast failure"`

Comment (scottkurz) [reply jakarta.batch.official.tck/src/main/asciidoc/batch-tck-reference-guide.adoc:334]: Sent you a PR fixing this and some pkg rename things.

**Comment by thadumi** [comment]
> we need to understand what settings you're using for style/formatting

this is my configuration for formatting. It's almost the one defined by JetBrains.
Here https://gist.github.com/thadumi/ffafd8f1273247cf30dd50e10b575baf you can find it.

Yesterday I was looking at the idea Martijn proposed. I like it but first I'll need to understand how to prepare the *.prefs files.

File: com.ibm.jbatch.tck.ann.proc/pom.xml (hunk 2 above)
Comment (thadumi) [reply com.ibm.jbatch.tck.ann.proc/pom.xml:20]: Regarding the copyright I'd open an issue for backtracking it and do the change with a different PR.

File: com.ibm.jbatch.tck.dist.exec/tck.execution.for.ri.properties (hunk 1 above)
Comment (thadumi) [reply com.ibm.jbatch.tck.dist.exec/tck.execution.for.ri.properties:58]: Yes, thanks for mentioning it. They should be 2.0.x and 3.0.x, shouldn't?

File: jakarta.batch.official.tck/src/main/asciidoc/batch-tck-reference-guide.adoc (hunk 4 above)
Comment (thadumi) [reply jakarta.batch.official.tck/src/main/asciidoc/batch-tck-reference-guide.adoc:349]: It's not going to create a new paragraph. A new paragraph is created only when there is an empty line between two sentences.

In this case I just split every sentence on it's own line.
To be honest I'd go even further and split every sentence in multiple lines. I find it quite hard to read the file with the current formatting.

**Comment by scottkurz** [comment]
Thx @thadumi for merging in my changes.   

The formatting questions could be revisited if/when we add the maven plugin enforcement...

Since you asked to move ahead with the other PR I thought I'd just go ahead and move this out of "Draft" status.. but I won't merge without running it by you.

I'm going to take one more review later tonight.  

---

On a separate note, Weld has released a snapshot:
https://github.com/starksm64/core/releases/tag/4.0.0-Alpha1

so I should be able to complete: 
https://github.com/WASdev/standards.jsr352.jbatch
to run the new TCK against it.  Will try to do more there later too.

**Comment by scottkurz** [comment]
> this is my configuration for formatting. It's almost the one defined by JetBrains.   Here https://gist.github.com/thadumi/ffafd8f1273247cf30dd50e10b575baf you can find it.

I think this would be OK to merge.  It doesn't mean we have to consider this the final, agreed-upon formatting.    It's an improvement though.

I don't see an easy way to use this format from an Eclipse IDE, though.  Maybe we should use something like the Google style that has representations in Intellij and Eclipse?   Though we could do later.  

I'd like to review the formatting change a bit more.

**Comment by scottkurz** [comment]
@thadumi thanks again for this.  I'm going to merge this.   

I tried your formatting in IDEA 2020.1 and it led to the same results except for:

* some formatting didn't get applied to comments (e.g. remove trailing whitespace).
* it didn't format the SQL /DDL files

We should probably talk more on the ML, since we should consider people using Eclipse, NetBeans, VSCode.  But I think it's better than what we had so I'm OK with it.
